agent/setup.go

```diff
@@ -101,6 +101,14 @@ func NewBaseDeps(configLoader ConfigLoader, logOut io.Writer, providedLogger hcl
 	cfg.Telemetry.PrometheusOpts.CounterDefinitions = counters
 	cfg.Telemetry.PrometheusOpts.SummaryDefinitions = summaries
 
+	if cfg.IsCloudEnabled() {
+		d.HCP, err = hcp.NewDeps(cfg.Cloud, d.Logger)
+		if err != nil {
+			return d, err
+		}
+	}
+	cfg.Telemetry.HCPSink = d.HCP.Sink
```

Comment:
Add the new HCPSink to the Telemetry config, so that it gets initialized in the line below `lib.InitTelemetry(cfg.Telemetry, d.Logger)`